### PR TITLE
DOC: update docstrings of signal.cspline2d, qspline2d, sepfir2d

### LINKED ISSUE #11928: DOC: signal: Wrong description for sepfir2d, cspline2d, qspline2d
My issue is about ...

The descriptions for sepfir2d, cspline2d, qspline2d looks wrong in the scipy.signal page:

SS 2020-04-24 at 21 27 14

#### Scipy/Numpy/Python version information:

DevDoc has this issue. 
- Signal processing (scipy.signal) — SciPy v1.5.0.dev0+4b3eac2 Reference Guide http://scipy.github.io/devdocs/signal.html

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix #11928

#### What does this implement/fix?
DOC: update docstrings of signal.cspline2d, qspline2d, sepfir2d

### CODE REVIEW

File: scipy/signal/splinemodule.c [hunk 1]
```diff
@@ -45,13 +45,33 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
 
 static char doc_cspline2d[] = "cspline2d(input {, lambda, precision}) -> ck\n"
 "\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:47]: The signature still doesn't show up in the doc for all three functions you're touching here. You could fix that by getting the format right here. See for example `convolve2d`, which also has its docs in C and the signature does show up.

File: scipy/signal/splinemodule.c [hunk 2]
```diff
@@ -45,13 +45,33 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
...
-"    input grid for the two-dimensional input image.  The lambda argument\n" 
-"    specifies the amount of smoothing.  The precision argument allows specifying\n"
-"    the precision used when computing the infinite sum needed to apply mirror-\n"
-"    symmetric boundary conditions.\n";
+"    Return the third-order B-spline coefficients over a regularly spaced\n"
+"    input grid for the two-dimensional input image.\n"
+"\n"
+"    Parameters\n"
+"    ----------\n"
+"\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:55]: `\n` between parameters and returns aren't necessary

**Review by rgommers — COMMENTED** [review]
Looks good, thanks @AtsushiSakai. Nice catch on these functions being included in the tutorial. If you could also make the signatures show up, that'd be even better.

File: scipy/signal/splinemodule.c (hunk 1 above)
Comment (AtsushiSakai) [reply scipy/signal/splinemodule.c:47]: Thanks. I agree. I fixed it.

File: scipy/signal/splinemodule.c (hunk 2 above)
Comment (AtsushiSakai) [reply scipy/signal/splinemodule.c]: I removed these.

File: scipy/signal/splinemodule.c [hunk 3]
```diff
@@ -168,14 +206,35 @@ static PyObject *qspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
...
+"    Parameters\n"
+"    ----------\n"
+"    input : ndarray\n"
+"        The input signal.\n"
+"\n"
+"    hrow : ndarray\n"
+"        A row direction filter defined by the rank-1 arrays.\n"
+"\n"
+"    hcol :ndarray\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:226]: minor: missing space after `:`

File: scipy/signal/splinemodule.c [hunk 4]
```diff
@@ -168,14 +206,35 @@ static PyObject *qspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
...
+"        A columun direction filter defined by the rank-1 arrays.\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:227]: typo `columun`

File: scipy/signal/splinemodule.c [hunk 5]
```diff
@@ -105,15 +124,34 @@ static PyObject *cspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
...
-"    specifies the amount of smoothing.  The precision argument allows specifying\n"
-"    the precision used when computing the infinite sum needed to apply mirror-\n"
-"    symmetric boundary conditions.\n";
+"    input grid for the two-dimensional input image.\n"
+"\n"
+"    Parameters\n"
+"    ----------\n"
+"    input : ndarray\n"
+"        The input signal.\n"
+"\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:138]: still `\n`'s here

File: scipy/signal/splinemodule.c [hunk 6]
```diff
@@ -105,15 +124,34 @@ static PyObject *cspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
...
+"    lambda : scalar\n"
+"        Specifies the amount of smoothingParameters in the transfer function.\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:140]: `smoothingParameters` -> `smoothing`?

File: scipy/signal/splinemodule.c [hunk 7]
```diff
@@ -105,15 +124,34 @@ static PyObject *cspline2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
...
+"\n"
+"    precision :\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:142]: missing type specifier, should be `float` it looks like

File: scipy/signal/splinemodule.c [hunk 8]
```diff
@@ -43,15 +43,34 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
...
-"    the precision used when computing the infinite sum needed to apply mirror-\n"
-"    symmetric boundary conditions.\n";
+"    Return the third-order B-spline coefficients over a regularly spaced\n"
+"    input grid for the two-dimensional input image.\n"
+"\n"
+"    Parameters\n"
+"    ----------\n"
+"    input : ndarray\n"
+"        The input signal.\n"
+"\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:57]: more `\n`'s here

File: scipy/signal/splinemodule.c [hunk 9]
```diff
@@ -43,15 +43,34 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
...
+"    lambda : scalar\n"
+"        Specifies the amount of smoothingParameters in the transfer function.\n"
+"\n"
+"    precision :\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:61]: same comment here, `: float`

File: scipy/signal/splinemodule.c [hunk 10]
```diff
@@ -43,15 +43,34 @@ convert_strides(npy_intp* instrides,npy_intp* convstrides,int size,int N)
...
+"    Return the third-order B-spline coefficients over a regularly spaced\n"
+"    input grid for the two-dimensional input image.\n"
+"\n"
+"    Parameters\n"
+"    ----------\n"
+"    input : ndarray\n"
+"        The input signal.\n"
+"\n"
+"    lambda : scalar\n"
+"        Specifies the amount of smoothingParameters in the transfer function.\n"
```
Comment (rgommers) [inline scipy/signal/splinemodule.c:59]: same comment on `smoothingParameters` -> `smoothing`

**Review by rgommers — COMMENTED** [review]
A number of very small tweaks still needed.

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
LGTM now, merged. Thanks @AtsushiSakai!

**Comment by sethtroisi** [comment]
Thanks for this improved documentation. I just referenced it today and found it much better than the pre-existing documentation which is still up on docs.scipy.org

**Pull request merged**